### PR TITLE
ENG-2549 Fix governance threshold: use >= instead of >

### PR DESCRIPTION
**Pull request by mividtim**
## Summary
- Fix governance threshold comparison in `_percentOwner`: changed `>` to `>=` so that owners holding exactly the threshold percentage can perform governance actions
- Without this fix, an owner with exactly 70% of supply is blocked from updating the operating agreement or validator, and exactly 50% is blocked from updating configuration — creating a potential governance deadlock

## Changes
- `src/FabricaToken.sol`: Changed `return percent > threshold` to `return percent >= threshold` in `_percentOwner`
- Updated 3 error messages and 3 comments from `>` to `>=` for consistency
- `test/FabricaTokenGovernance.t.sol`: 12 new Foundry tests covering all governance functions at boundary conditions

## Manual Anvil Fork Testing

Ran all 12 governance tests against a local Anvil fork of Base Sepolia to verify on-chain behavior:

```
forge test --match-contract FabricaTokenGovernanceTest --fork-url http://127.0.0.1:8546 -vv

Ran 12 tests for test/FabricaTokenGovernance.t.sol:FabricaTokenGovernanceTest
[PASS] test_updateConfig_exactlyAt50Percent()          — 50/100 owner CAN update config (>= fix)
[PASS] test_updateConfig_above50Percent()               — 51/100 owner CAN update config
[PASS] test_updateConfig_below50Percent_reverts()       — 49/100 owner CANNOT update config
[PASS] test_updateConfig_roundingDown_reverts()         — 1/3 owner (33%) CANNOT update config
[PASS] test_updateOA_exactlyAt70Percent()               — 70/100 owner CAN update OA (>= fix)
[PASS] test_updateOA_above70Percent()                   — 71/100 owner CAN update OA
[PASS] test_updateOA_below70Percent_reverts()           — 69/100 owner CANNOT update OA
[PASS] test_updateOA_100PercentOwner()                  — 100% owner CAN update OA
[PASS] test_updateOA_zeroBalance_reverts()              — 0% owner CANNOT update OA
[PASS] test_updateOA_roundingDown_boundary()            — 7/10 owner (70%) CAN update OA
[PASS] test_updateValidator_exactlyAt70Percent()        — 70/100 owner CAN update validator (>= fix)
[PASS] test_updateValidator_below70Percent_reverts()    — 69/100 owner CANNOT update validator

Suite result: ok. 12 passed; 0 failed; 0 skipped
```

Key boundary verifications:
1. **Exactly 70% (70/100) CAN update OA** — was blocked by > bug, now works with >=
2. **69% (69/100) CANNOT update OA** — correctly reverts
3. **Exactly 50% (50/100) CAN update config** — was blocked, now works
4. **49% (49/100) CANNOT update config** — correctly reverts
5. **Math.mulDiv rounding**: 7/10 = 70% passes, 1/3 = 33% correctly fails

## Test plan
- [x] 12 Foundry unit tests covering all 3 governance functions at threshold boundaries
- [x] Anvil fork testing against Base Sepolia
- [x] Verified exact-threshold owners can now perform actions (bug fix confirmed)
- [x] Verified below-threshold owners still correctly revert (no regression)

## Summary by CodeRabbit

* **Bug Fixes**
  * Governance action thresholds are now inclusive (≥ instead of >), allowing actions at exactly the threshold ownership percentage: 70% for operating agreement and validator updates, 50% for configuration updates.

* **Tests**
  * Comprehensive governance behavior test suite added, covering exact thresholds, above/below thresholds, rounding, and edge cases.

### CODE REVIEW

File: test/FabricaTokenGovernance.t.sol [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+    }
+
+    function test_updateOA_roundingDown_boundary() public {
+        // Alice owns 7 out of 10. Math.mulDiv(7, 100, 10) = 70. 70 >= 70.
+        uint256 id = _mintSplit(7, 3);
+        vm.prank(alice);
+        bool result = token.updateOperatingAgreement("https://new-oa.com", id);
+        assertTrue(result, "7/10 ownership (70%) should pass >= 70% threshold");
+    }
+}
```
[the remark this answers is not in the document]
Comment (mividtim) [reply test/FabricaTokenGovernance.t.sol:170]: Fixed — ran `forge fmt` and pushed in 5e9e67d. Going forward, will always run `forge fmt` before pushing.

File: test/FabricaTokenGovernance.t.sol [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+        assertTrue(result, "Exactly 70% owner should be able to update validator");
+    }
+
+    function test_updateValidator_below70Percent_reverts() public {
+        // Alice owns 69 out of 100 (69%)
+        uint256 id = _mintSplit(69, 31);
+        vm.prank(alice);
+        vm.expectRevert("Only >= 70% can update");
+        token.updateValidator(address(0xDEAD), id);
+    }
```
[the remark this answers is not in the document]
Comment (mividtim) [reply test/FabricaTokenGovernance.t.sol:151]: Added `test_updateValidator_above70Percent` in 14264c2 — good catch for consistency.

File: test/FabricaTokenGovernance.t.sol [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+    function defaultOperatingAgreement() external pure returns (string memory) {
+        return "https://example.com/oa";
+    }
+
+    function operatingAgreementName(string memory) external pure returns (string memory) {
+        return "Default OA";
+    }
+
+    function uri(uint256) external pure returns (string memory) {
+        return "https://example.com/uri";
```
[the remark this answers is not in the document]
Comment (mividtim) [reply test/FabricaTokenGovernance.t.sol]: Good catch — added `override` to all three MockValidator methods in e8ec63c.

File: test/FabricaTokenGovernance.t.sol [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+        assertTrue(result, "71% owner should be able to update validator");
+    }
+
+    function test_updateValidator_below70Percent_reverts() public {
+        // Alice owns 69 out of 100 (69%)
+        uint256 id = _mintSplit(69, 31);
+        vm.prank(alice);
+        vm.expectRevert("Only >= 70% can update");
+        token.updateValidator(address(0xDEAD), id);
+    }
```
[the remark this answers is not in the document]
Comment (mividtim) [reply test/FabricaTokenGovernance.t.sol:151]: Good suggestion — switched all three `updateValidator` tests to use `address(validator)` instead of `address(0xDEAD)` in e8ec63c. More resilient if validation is added later.

**Comment by mividtim** [comment]
Tim approved this for merge.
